raiden/network/proxies/token_network.py

```diff
@@ -814,94 +800,104 @@ def _set_total_deposit(
+                    msg = (
+                        f"setTotalDeposit failed and all gas was used "
+                        f"({gas_limit}). Estimate gas may have underestimated "
+                        f"setTotalDeposit, or succeeded even though an assert is "
+                        f"triggered, or the smart contract code has an "
+                        f"conditional assert."
+                    )
+                    raise RaidenUnrecoverableError(msg)
 
-            raise error_type(f"{error_prefix}. {msg}")
+                # Query the current state to check for transaction races
+                sender_details = self._detail_participant(
+                    channel_identifier=channel_identifier,
+                    detail_for=self.node_address,
+                    partner=partner,
+                    block_identifier=given_block_identifier,
+                )
 
-    def _check_why_deposit_failed(
-        self,
-        channel_identifier: ChannelID,
-        partner: Address,
-        token: Token,
-        amount_to_deposit: TokenAmount,
-        total_deposit: TokenAmount,
-        transaction_executed: bool,
-        block_identifier: BlockSpecification,
-    ) -> Tuple[ErrorType, str]:
-        error_type: ErrorType = RaidenUnrecoverableError
-        msg = ""
-        latest_deposit = self._detail_participant(
-            channel_identifier=channel_identifier,
-            detail_for=self.node_address,
-            partner=partner,
-            block_identifier=block_identifier,
-        ).deposit
+                total_deposit_done = sender_details.deposit >= total_deposit
+                if total_deposit_done:
+                    raise RaidenRecoverableError("Requested total deposit was already performed")
 
-        allowance = token.allowance(
-            owner=self.node_address,
-            spender=Address(self.address),
-            block_identifier=block_identifier,
-        )
-        if allowance < amount_to_deposit:
-            msg = (
-                "The allowance is insufficient. Check concurrent deposits "
-                "for the same token network but different proxies."
-            )
-        elif token.balance_of(self.node_address, block_identifier) < amount_to_deposit:
-            msg = "The address doesnt have enough tokens"
-        elif transaction_executed and latest_deposit < total_deposit:
-            msg = "The tokens were not transferred"
+                failed_at = self.proxy.jsonrpc_client.get_block("latest")
+                failed_at_blockhash = encode_hex(failed_at["hash"])
+                latest_deposit = self._detail_participant(
+                    channel_identifier=channel_identifier,
+                    detail_for=self.node_address,
+                    partner=partner,
+                    block_identifier=failed_at_blockhash,
+                ).deposit
+                if latest_deposit < total_deposit:
```

Comment:
This check is not correct as it stands. It is perfectly fine for the deposit to stay the same, if the channel was closed before the set_total_deposit was mined. Please add the check for the channel state before this.